packages/libraries/core/src/client/cdn-artifact-fetcher.ts

```diff
@@ -0,0 +1,171 @@
+import { Logger } from '@graphql-hive/logger';
+import CircuitBreaker from '../circuit-breaker/circuit.js';
+import { version } from '../version.js';
+import {
+  CircuitBreakerConfiguration,
+  defaultCircuitBreakerConfiguration,
+} from './circuit-breaker.js';
+import { http, HttpCallConfig } from './http-client.js';
+import { chooseLogger, createHash } from './utils.js';
+
+type CreateCDNArtifactFetcherArgs = {
+  /**
+   * The endpoint that should be fetched.
+   *
+   * It is possible to provide an endpoint list. The first endpoint will be treated as the primary source.
+   * The secondary endpoint will be used in case the first endpoint fails to respond.
+   *
+   * Example:
+   *
+   * ```
+   * [
+   *          "https://cdn.graphql-hive.com/artifacts/v1/9fb37bc4-e520-4019-843a-0c8698c25688/supergraph",
+   *   "https://cdn-mirror.graphql-hive.com/artifacts/v1/9fb37bc4-e520-4019-843a-0c8698c25688/supergraph"
+   * ]
+   * ```
+   */
+  endpoint: string | [string, string];
```

Comment:
Not sure it is necessary to be this strict with the type :-) The code is already working for whatever array length :-) And it avoids casting when using `.map` or `.filter`!
```suggestion
  endpoint: string | string[];
```